src/views/nads/form/utils/utils.ts

```diff
@@ -0,0 +1,116 @@
+import { adjectives, animals, uniqueNamesGenerator } from 'unique-names-generator';
+
+import NetworkAttachmentDefinitionModel from '@kubevirt-ui/kubevirt-api/console/models/NetworkAttachmentDefinitionModel';
+import { k8sCreate } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  IPAMConfig,
+  NetworkAttachmentDefinitionAnnotations,
+  NetworkAttachmentDefinitionConfig,
+  NetworkAttachmentDefinitionKind,
+  RESOURCE_NAME_ANNOTATION,
+} from '@utils/resources/nads/types';
+import { isEmpty } from '@utils/utils';
+
+import { NetworkAttachmentDefinitionFormInput, NetworkTypeKeys } from './types';
+import { networkConsole } from '@utils/utils/utils';
+
+export const generateNADName = (): string => {
+  return `network-${uniqueNamesGenerator({
+    dictionaries: [adjectives, animals],
+    separator: '-',
+  })}`;
+};
+
+const buildConfig = (
+  formData: NetworkAttachmentDefinitionFormInput,
+  namespace: string,
+): NetworkAttachmentDefinitionConfig => {
+  const { name, networkType } = formData;
+  const networkTypeData = formData?.[networkType];
+
+  const commonConfig: Partial<NetworkAttachmentDefinitionConfig> = {
+    cniVersion: '0.3.1',
+    name,
+    type: networkType,
+  };
+
+  const ipam = parseIPAM(networkTypeData?.ipam);
+  const netAttachDefName = `${namespace}/${name}`;
+
+  const specificConfig: Record<NetworkTypeKeys, Partial<NetworkAttachmentDefinitionConfig>> = {
+    [NetworkTypeKeys.cnvBridgeNetworkType]: {
+      bridge: networkTypeData?.bridge,
+      ipam,
+      macspoofchk: networkTypeData?.macspoofchk,
+      preserveDefaultVlan: false,
+      vlan: parseInt(networkTypeData?.vlanTagNum, 10) || undefined,
+    },
+    [NetworkTypeKeys.ovnKubernetesNetworkType]: {
+      netAttachDefName,
+      topology: 'layer2',
+    },
+    [NetworkTypeKeys.ovnKubernetesSecondaryLocalnet]: {
+      cniVersion: '0.4.0',
+      mtu: parseInt(networkTypeData?.mtu, 10) || undefined,
+      name: networkTypeData?.bridgeMapping,
+      netAttachDefName,
+      topology: 'localnet',
+      vlanID: parseInt(networkTypeData?.vlanID, 10) || undefined,
+    },
+    [NetworkTypeKeys.sriovNetworkType]: {
+      ipam,
+    },
+  };
+
+  return { ...commonConfig, ...specificConfig[networkType] };
+};
+
+const parseIPAM = (ipamString: string | undefined): IPAMConfig => {
+  try {
+    return JSON.parse(ipamString || '{}');
+  } catch (e) {
+    networkConsole.error('Could not parse IP address management JSON', e);
+    return {};
+  }
+};
+
+const getResourceName = (
+  formData: NetworkAttachmentDefinitionFormInput,
+  networkType: string,
+): string => {
+  const paramsData = formData?.[networkType];
+
+  if (isEmpty(paramsData)) return null;
+
+  return networkType === NetworkTypeKeys.cnvBridgeNetworkType
+    ? `bridge.network.kubevirt.io/${paramsData?.bridge ?? ''}`
+    : `openshift.io/${paramsData?.resourceName ?? ''}`;
```

Comment:
both labels will be end with a / in case param.bridge or params.resource are undefined .. it will work?

Reply:
Well you're right but this is exactly how it used to be, so I have no clue how it should be.
Because of our validations from react-hook-form the first annotation will always end with a value since `networkType === NetworkTypeKeys.cnvBridgeNetworkType` enforces the value for bridge otherwise the submit is disabled so we can also omit the ?? '' part.
About the else section since resourceName is defined only when `networkType === NetworkTypeKeys.sriovNetworkType` the can get / at the end, I think I should check for if it's sriov type and submit with a value otherwise return null, what do you think? @metalice

Reply:
if i remember correctly, there was a bug regarding this issue of ending in / , the solution was to remove the label if there is no value after the /